### PR TITLE
RavenDB-6226

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

**Comment by aviviadi** [comment]
http://issues.hibernatingrhinos.com/issue/RavenDB-6226

File: src/Raven.Server/Documents/TransactionOperationsMerger.cs [hunk 1]
```diff
@@ -183,9 +183,15 @@ private void MergeTransactionsOnce()
...
                         case PendingOperations.ModifiedsSystemDocuments:
-                            tx.Commit();
-                            tx.Dispose();
-                            NotifyOnThreadPool(pendingOps);
+                            try
+                            {
+                                tx.Commit();
+                                tx.Dispose();
+                            }
+                            finally
```
Comment (ayende) [inline src/Raven.Server/Documents/TransactionOperationsMerger.cs:191]: What sets the op.Exception here? Are we sure we won't be sending an ok when we actually errored?

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 2]
```diff
@@ -287,8 +301,8 @@ private void RecoverCurrentJournalSize(AbstractPager pager)
                 return null;
             }
 
-            // write transactions can read directly from journals that they got when they started up
-            var files = tx.JournalFiles;
+            // write transactions can read directly from journals
+            var files = _files;
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs:305]: Why did you do this? This reproduce the race condition we fixed.

File: src/Voron/Platform/Posix/PosixMemoryMapPager.cs [hunk 3]
```diff
@@ -206,6 +206,12 @@ public override void ReleaseAllocationInfo(byte* baseAddress, long size)
             NativeMemory.UnregisterFileMapping(FileName, ptr, size);
         }
 
+        public override void FillZeroLengthJournalWithZeros(ulong size)
```
Comment (ayende) [inline src/Voron/Platform/Posix/PosixMemoryMapPager.cs:209]: Will not actually work, the error is actually being thrown from the constructor.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Voron/Impl/Journal/WriteAheadJournal.cs (hunk 2 above)
Comment (aviviadi) [reply src/Voron/Impl/Journal/WriteAheadJournal.cs:305]: I didn't do specifically that.. It was from your old pull i guess

**Pull request merged**